libs/entitlement/qa/entitled-plugin/src/main/java/org/elasticsearch/entitlement/qa/entitled/EntitledPlugin.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.qa.entitled;
+
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.entitlement.runtime.api.NotEntitledException;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.plugins.ExtensiblePlugin;
+import org.elasticsearch.plugins.Plugin;
+
+public class EntitledPlugin extends Plugin implements ExtensiblePlugin {
+
+    @SuppressForbidden(reason = "Exposes forbidden APIs for testing purposes")
+    public static void System_clearProperty(String key) {
```

Comment:
Any reason this is public?

Reply:
The notion is that this is a method that would be called by a test case that wants to clear a system property. It's a bit artificial because this PR only contains the new plugin; use cases will be in the next PR.

In all likelihood I'll delete this method at that point. In fact, `selfTest` might even go away once we have real uses in real tests.

Reply:
Could we remove these in this PR and leave this for future changes to utilize? I see this PR as the infra, but we don't need an artificial usecase.

Reply:
I couldn't tell whether the design was suitable with out some use case. PRs with infra and no use case are weird. I'm still getting use to this practice.

But yeah, I could make that method private if it bugs you guys. 😄

Reply:
It can't be private now because it's in another class. It could be package-private, but I think the very next PR is going to delete it anyway.